util/src/main/java/chat/rocket/android/util/extension/Image.kt

```diff
@@ -96,6 +97,17 @@ fun String.getCompressFormat(): Bitmap.CompressFormat {
     }
 }
 
+/**
+ * Returns true if lossy compressible format based on mimeType
+ */
+fun String.lossyCompressible(): Boolean {
+    return when {
```

Comment:
Since this function is only invoked when uploading an image from the camera (see comment above), and the camera on my emulator is creating png's, if the image happens to be too large, then this check will throw an exception and the image will never get compressed.